api/src/utils.ts

```diff
@@ -0,0 +1,64 @@
+import { MiddlewareHandler } from "hono";
+
+export const entries = <T extends object>(o: T) => {
+	return Object.entries(o) as [keyof T, T[keyof T]][];
+};
+export const filterObject = <T extends object>(
+	o: T,
+	predicate: (o: [keyof T, T[keyof T]]) => boolean,
+) => {
+	return Object.fromEntries(entries(o).filter(([key, value]) => predicate([key, value])));
+};
+
+export const mapObject = <T extends object, R>(
+	o: T,
+	func: (o: [keyof T, T[keyof T]]) => [keyof T, R],
+): Record<keyof T, R> => {
+	return Object.fromEntries(entries(o).map(([k, v]) => func([k, v]))) as Record<keyof T, R>;
+};
+
+export const languagesFilter =
+	(nested: boolean): MiddlewareHandler =>
+	async (c, next) => {
+		const languages = c.req.query("lang")?.split(",") ?? ["en"];
```

Comment:
Point of curiosity: Are we able to fetch the user's configured language from the browser and use that as a primary fallback, with english as a secondary fallback? This is an API, so maybe not, but it's worth asking.

Reply:
Requests from the API could be run on the server, where a locale isn't available, and it's better to keep things consistent and only rely on inputs from the actual request.